claim_batch/services.py

```diff
@@ -431,6 +432,109 @@ def _get_capitation_region_and_district(location_id):
 def do_process_batch(audit_user_id, location_id, period, year):
     processed_ids = set()  # As we update claims, we add the claims not in relative pricing and then update the status
     logger.debug("do_process_batch location %s for %s/%s", location_id, period, year)
+    batch_run = new BatchRun
+
+    # 0 prepare the batch run :  does it really make sense per location ? (Ideally per pool but the notion doesn't exist yet)
+    # 0.1 get all product concerned, all product that have are configured for the location
+    _, days_in_month = calendar.monthrange(year, period)
+    end_date = datetime.date(year, period, days_in_month)
+    period_quarter = period - 2 if period|divisibleby:4 else 0
+    period_sem = period - 5 if period|divisibleby:6 else 0
+
+    products = Product.objects\
+            .filter(validity_from__lte = start_date)\
```

Comment:
@delcroip 'start_date' is not defined in that scope of code. Could be 
    ```
_, days_in_month = calendar.monthrange(year, period)
    start_date = datetime.date(year, period, 1)
    end_date = datetime.date(year, period, days_in_month)
    period_quarter = period - 2 if period|4 else 0
    period_sem = period - 5 if period|6 else 0
```
?

Reply:
I was sure it was there ... Yes do it

Reply:
ok